crates/runtime/src/starknet/context.rs

```diff
@@ -140,6 +143,13 @@ pub fn build_context(block_info: BlockInfo) -> EntryPointExecutionContext {
     .unwrap()
 }
 
+pub fn set_max_steps(entry_point_ctx: &mut EntryPointExecutionContext, max_n_steps: u32) {
+    entry_point_ctx.block_context.invoke_tx_max_n_steps = max_n_steps;
```

Comment:
Why is this one needed? isn't `vm_run_resources` sufficient?

Reply:
I followed the logic of `build_context()` -> `build_block_context()` to get same state that with `BlockContext` direct construction.
But yes, in fact it's not necessary.

Reply:
I would leave the line though, I think it's better to be consistent xD